File: BinaryTree/BinaryTreeRightSideView.js
/*
Given a binary tree, imagine yourself standing on the right side of it, return the values of the nodes you can see ordered from top to bottom.

For example:
Given the following binary tree,
   1            <---
 /   \
2     3         <---
 \     \
  5     4       <---
You should return [1, 3, 4].

*/

/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var rightSideView = function(root) {
    var search = function(n, level, res){
        if(n === null) return;
        if(level > res.length) res.push(n.val);
        search(n.right, level + 1, res);
        search(n.left, level + 1, res);
    }
    var res = [];
    search(root, 1, res);
    return res;
};

File: BinaryTree/BinaryTreeRightSideView.java
/*
Given a binary tree, imagine yourself standing on the right side of it, return the values of the nodes you can see ordered from top to bottom.

For example:
Given the following binary tree,
   1            <---
 /   \
2     3         <---
 \     \
  5     4       <---
You should return [1, 3, 4].

*/

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> list = new ArrayList<Integer>();
        search(root, 1, list);
        return list;
    }
    private void search(TreeNode n, int level, List<Integer> list){
        if(n == null) return;
        if(level > list.size()) list.add(n.val);
        search(n.right, level + 1, list);
        search(n.left, level + 1, list);
    }
}